src/library_pipefs.js

```diff
@@ -79,8 +79,7 @@ mergeInto(LibraryManager.library, {
         });
 
         assert(buffer instanceof ArrayBuffer || ArrayBuffer.isView(buffer));
-        var data = new Uint8Array(buffer);
-        data = buffer.subarray(offset, offset + length);
+        var data = buffer.subarray(offset, offset + length);
```

Comment:
The previous code (and similar code in `write`) looks like an incorrectly written conversion to the byte-sized buffer. Is it safe to assume that the `buffer` is always already byte-sized?